src/cargo/core/compiler/job_queue/mod.rs

```diff
@@ -314,8 +315,10 @@ impl<'cfg> DiagDedupe<'cfg> {
             return Ok(false);
         }
         let mut shell = self.config.shell();
-        shell.print_ansi_stderr(diag.as_bytes())?;
-        shell.err().write_all(b"\n")?;
+        if shell.warnings() != WarningHandling::Ignore {
+            shell.print_ansi_stderr(diag.as_bytes())?;
+            shell.err().write_all(b"\n")?;
```

Comment:
Maybe I missed it but how are we turning compiler warnings into errors?

Reply:
We're not turning the warnings into errors. They're still printed as warnings.

After all the warnings are emitted, a single error is emitted if any warnings were present.